controllers/Oauth.spec.js

```diff
@@ -0,0 +1,38 @@
+const server = require('../api/server');
+// eslint-disable-next-line import/order
+const request = require('supertest')(server);
+const Oauth = require('./Oauth');
+
+const { socialAuth, getAuthToken } = Oauth;
+
+describe('Oauth.js Tests', () => {
+  describe('socialAuth function', () => {
+    test('Returns a promise', async () => {
+      const req = {};
+      const res = {
+        status(statusCode) {
+          return {
+            json() {}
+          };
+        }
+      };
+      const isPromise = typeof socialAuth(req, res).then === 'function';
```

Comment:
Doesn't this currently check if it returns a function rather than a Promise? Might prove to send a false positive.

Reply:
A Promise should have a then and catch function. This just checks if there is a then function

Reply:
Alright makes sense, thanks!